### PR TITLE
Cleanin' up stuff

### PR DESCRIPTION
**Pull request by michaelbprice**
#### PR Classification
Code cleanup and minor enhancements.

#### PR Summary
This pull request includes code cleanup, memory alignment adjustments, and minor enhancements.
- `CMakeLists.txt`: Removed `imgui_demo.cpp` and added a preprocessor macro definition.
- `Input.h`: Updated `JoystickInfo::Axis` struct to use `float` literals.
- `SystemEditor.cpp`: Removed several unnecessary `#include` directives.
- `StarSystem.h`: Added `#pragma pack` directives for specific memory alignment.
- `TestStringName.cpp`: Added a new `SUBCASE` to test default construction and properties of `StringName`.

### CODE REVIEW

File: src/Input.h [hunk 1]
```diff
@@ -76,12 +76,12 @@ namespace Input {
 	};
 
 	struct JoystickInfo {
-		struct Axis {
-			float value = 0.0;
-			float deadzone = 0.1;
-			float curve = 1.0;
-			bool zeroToOne = false;
-		};
+        struct Axis {
```
Comment (MryamGirmay) [inline src/Input.h:79]: Thank you for resolving the initialization warning.

File: src/galaxy/StarSystem.h [hunk 2]
```diff
@@ -24,7 +24,7 @@ class CustomSystem;
 
 // doubles - all masses in Kg, all lengths in meters
 // fixed - any mad scheme
-
+#pragma pack(push, 4)
```
Comment (MryamGirmay) [inline src/galaxy/StarSystem.h:27]: Great job on reducing memory padding here.

Comment (michaelbprice) [reply src/galaxy/StarSystem.h:27]: I agree.

File: src/Input.h (hunk 1 above)
Comment (michaelbprice) [reply src/Input.h:79]: np

**Pull request closed without merging**